extension/src/popup/views/SignTransaction/index.tsx

```diff
@@ -87,11 +140,49 @@ export const SignTransaction = () => {
     }
   }, [isMemoRequired, isMalicious, isUnsafe]);
 
-  const isSubmitDisabled = isMemoRequired || isMalicious;
+  useEffect(() => {
+    // handle auto selecting the right account based on `accountToSign`
+    let currentAccountDetails;
+    let defaultAccountDetails;
 
-  const { networkName, otherNetworkName, networkPassphrase } = useSelector(
-    settingsNetworkDetailsSelector,
-  );
+    allAccounts.forEach((account) => {
```

Comment:
I wanted to only loop over allAccount once and get all the info needed in one go. This makes it slightly more complex so let me know if this is too confusing now!